llvm/lib/Target/AIE/AIECombinerHelper.cpp

```diff
@@ -581,9 +593,14 @@ static bool canProduceS20(const MachineRegisterInfo &MRI,
   case TargetOpcode::G_CONSTANT:
   case TargetOpcode::G_IMPLICIT_DEF:
     return true;
-  default:
+  default: {
+    // Since generic extract vector element opcodes are determined at runtime,
+    // we call a function in if statement to handle them.
+    if (IsGenericExtractVectorElt(MI))
+      return true;
```

Comment:
can we do the following, instead of putting directly in default ? 

```
case TargetOpcode::G_INTRINSIC
    if (IsGenericExtractVectorElt(MI))
      return true;
```

Reply:
Same as above